tardis/visualization/tools/sdec_plot.py

```diff
@@ -1178,6 +1181,33 @@ def generate_plot_mpl(
                 linewidth=1,
             )
 
+        # Plot observed spectrum
+        if observed_spectrum:
+            observed_spectrum_wavelength = None
+            observed_spectrum_luminosity = None
+
+            if (
+                type(observed_spectrum[0]) is u.quantity.Quantity
+                and type(observed_spectrum[1]) is u.quantity.Quantity
+            ):
+                # If the observed spectrum is a list of quantities, then convert to wavelength and luminosity units
+                observed_spectrum_wavelength = observed_spectrum[0].to(u.AA)
+                observed_spectrum_luminosity = observed_spectrum[1].to(
+                     "erg / s"
+                )
+
+            else:
+                raise ValueError(
+                    "Both wavelength and luminosity should be astropy quantities."
+                )
```

Comment:
I don't think we need to do type checking unless necessary (search on internet why - it's a good thing to know). `.to` function will automatically fail if it's not a quantity or has incompatible units